sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnarBatchSuite.scala

```diff
@@ -739,6 +739,123 @@ class ColumnarBatchSuite extends SparkFunSuite {
     }}
   }
 
+  test("Nest Array in Array.") {
```

Comment:
Can we prepare a test for primitive type array (e.g. `new ArrayType(new ArrayType(IntegerType, false), true)`?